### PR TITLE
Create display of summary view of match

### PR DESCRIPTION
**Pull request by kubaolszewski**

### CODE REVIEW

File: lib/app/custom_widgets/live_match_tile.dart [hunk 1]
```diff
@@ -1,99 +1,186 @@
...
-          color: AppColors.listTileGrey,
-          borderRadius: BorderRadius.circular(12),
-        ),
-        child: Padding(
-          padding: const EdgeInsets.all(8.0),
-          child: Column(
-            mainAxisAlignment: MainAxisAlignment.spaceAround,
-            children: [
-              Row(
+    double width = MediaQuery.of(context).size.width;
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Not sure if that's the best solution.

File: lib/app/custom_widgets/live_match_tile.dart [hunk 2]
```diff
@@ -1,99 +1,186 @@
...
+    final assetName = liveMatch.league!.flag;
+    const String defaultFlag =
+        'https://thumbs.dreamstime.com/b/handshake-vector-icon-black-illustration-isolated-graphic-web-design-business-contract-agreement-flat-symbol-white-98077091.jpg';
+    return Container(
+      width: width,
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: I would use `double.infinity`

File: lib/app/custom_widgets/live_match_tile.dart [hunk 3]
```diff
@@ -1,99 +1,186 @@
...
+                                  ),
+                                ),
+                        ),
                       ),
-                      SizedBox(width: 10),
+                      const SizedBox(width: 10),
                       Text(
-                        'League name',
-                        style: TextStyle(
+                        liveMatch.league!.name!,
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Are you sure it will never be null?

File: lib/app/custom_widgets/live_match_tile.dart [hunk 4]
```diff
@@ -1,99 +1,186 @@
...
                       color: AppColors.liveTimerBackground,
                     ),
-                    child: const Row(
+                    child: Row(
                       mainAxisAlignment: MainAxisAlignment.spaceAround,
                       children: [
                         CircleAvatar(
                           radius: 5,
-                          backgroundColor: Colors.green,
+                          backgroundColor: liveMatch.fixture!.status!.short! == 'TBD' ||
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Are you sure it will never be null?

File: lib/app/custom_widgets/live_match_tile.dart [hunk 5]
```diff
@@ -1,99 +1,186 @@
...
+                  children: [
+                    Expanded(
+                      flex: 1,
+                      child: SizedBox.square(
+                        dimension: 110,
+                        child: Column(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+                            Image(
+                              image: NetworkImage(liveMatch.teams!.home!.logo!, scale: 3),
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/live_match_tile.dart]: Are you sure it will never be null?

File: lib/app/custom_widgets/wide_match_tile.dart [hunk 6]
```diff
@@ -1,110 +1,140 @@
...
-                          color: Colors.grey,
-                          fontSize: 16,
-                          fontWeight: FontWeight.bold,
-                        ),
+                      Row(
+                        children: [
+                          Image(
+                            width: 35,
+                            height: 35,
+                            image: NetworkImage(liveMatch.teams!.home!.logo!),
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/wide_match_tile.dart]: Same here

File: lib/app/custom_widgets/wide_match_tile.dart [hunk 7]
```diff
@@ -1,110 +1,140 @@
...
+                          )
+                        ],
                       ),
+                      const SizedBox(height: 10),
+                      Row(
+                        children: [
+                          Image(
+                            width: 35,
+                            height: 35,
+                            image: NetworkImage(liveMatch.teams!.away!.logo!),
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/wide_match_tile.dart]: Same here

File: lib/app/features/home/home.dart
Comment (MBanaszynski) [inline lib/app/features/home/home.dart:1]: I would recommend to rename file to `home_screen.dart`

File: lib/app/features/home/home.dart [hunk 8]
```diff
@@ -1,91 +1,110 @@
...
+import 'package:score_live/app/features/home/home_tabs/upcoming_tab/upcoming_tab.dart';
+import 'package:score_live/app/features/home/home_tabs/score_tab/score_tab.dart';
+import 'package:score_live/app/features/home/home_tabs/favorites_tab/favorites_tab.dart';
 import 'package:score_live/core/applocalization_context.dart';
 import 'package:score_live/core/enums.dart';
 import 'package:score_live/presentation/constants/app_colors.dart';
+import 'package:score_live/presentation/constants/common_text_styles.dart';
 
-class HomeScreen extends StatelessWidget {
+class HomeScreen extends StatefulWidget {
```
Comment (MBanaszynski) [inline lib/app/features/home/home.dart]: I would leave `HomeScreen` as `StatelessWidget` and save only bloc provider in this file. The rest of the view I would move to separated file e.g. `home_view.dart`

File: lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart [hunk 9]
```diff
@@ -9,18 +8,17 @@ import 'package:score_live/presentation/constants/app_colors.dart';
 class HomeOptionsTapBar extends StatelessWidget {
   const HomeOptionsTapBar({super.key});
 
-  HomeCubit _homeCubit(BuildContext context) => context.read<HomeCubit>();
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart]: Why did you remove this?

File: lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart [hunk 10]
```diff
@@ -9,18 +8,17 @@ import 'package:score_live/presentation/constants/app_colors.dart';
...
-
   @override
   Widget build(BuildContext context) {
+    final homeCubit = Modular.get<HomeCubit>();
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart]: The previous solution was way better

File: lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart [hunk 11]
```diff
@@ -1,4 +1,3 @@
-
 import 'package:flutter/material.dart';
 import 'package:flutter_modular/flutter_modular.dart';
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart]: Add `hide ModularWatchExtension`

File: lib/app/features/home/home_widgets/home_screen_date_picker/home_screen_date_picker.dart [hunk 12]
```diff
@@ -0,0 +1,60 @@
+import 'package:date_picker_timeline/date_picker_widget.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:score_live/app/features/home/cubit/home_cubit.dart';
+import 'package:score_live/app/features/home/home_tabs/score_tab/cubit/score_tab_cubit.dart';
+import 'package:score_live/core/applocalization_context.dart';
+import 'package:score_live/presentation/constants/app_colors.dart';
+
+class HomeScreenDatePicker extends StatefulWidget {
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/home_screen_date_picker/home_screen_date_picker.dart]: Do you need this to be `StatefulWidget` ?

File: lib/app/features/home/match_details/cubit/match_details_cubit.dart [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+part 'match_details_state.dart';
+part 'match_details_cubit.freezed.dart';
+
+class MatchDetailsCubit extends Cubit<MatchDetailsState> {
+  MatchDetailsCubit(this.matchDetailsRepository) : super(const MatchDetailsState());
+
+  final MatchDetailsRepository matchDetailsRepository;
+
+  Future<void> fetchMatchEvents(
+    // String matchID,
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/cubit/match_details_cubit.dart]: Remove this, if not needed

File: lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart [hunk 14]
```diff
@@ -12,8 +22,67 @@ class MatchSummaryView extends StatelessWidget {
...
+                        height: 200,
+                        child: ListView(
+                          physics: const NeverScrollableScrollPhysics(),
+                          children: [
+                            for (final event in matchEvents) ...[
+                              if (liveMatch.teams!.home!.id == event.team!.id && event.time!.elapsed! < 45)
+                                Padding(
+                                  padding: const EdgeInsets.only(left: 8.0, top: 8.0),
+                                  child: HomeEventTextSample(
+                                      time: event.time!.elapsed.toString(), player: event.player!.name!),
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart]: Are you sure it will never be null?

File: lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart [hunk 15]
```diff
@@ -29,22 +98,124 @@ class MatchSummaryView extends StatelessWidget {
...
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: [
+                        liveMatch.score!.halftime!.home == null || liveMatch.score!.halftime!.away == null
+                            ? Text(
+                                liveMatch.fixture!.status!.long!,
+                                textAlign: TextAlign.center,
+                                style: const TextStyle(color: Colors.white, fontSize: 16),
+                              )
+                            : Text(
+                                '${liveMatch.teams!.home!.name} ${liveMatch.score!.halftime!.home} - '
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart]: Same here

File: lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart:1]: Reorganise this file please, move sections to separated files

**Review by MBanaszynski — CHANGES_REQUESTED** [review]

File: lib/app/features/home/home_widgets/home_screen_date_picker/home_screen_date_picker.dart (hunk 12 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_widgets/home_screen_date_picker/home_screen_date_picker.dart]: fixed

File: lib/app/features/home/home.dart (hunk 8 above)
Comment (kubaolszewski) [reply lib/app/features/home/home.dart]: done

File: lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_screens.dart/match_summary_view.dart:1]: done

File: lib/app/custom_widgets/match_league_and_timer_widget.dart [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: [
+              CircleAvatar(
+                radius: 5,
+                backgroundColor: matchStatusShort == 'TBD' || matchStatusShort == 'NS' || matchStatusShort == 'SUSP'
+                    ? Colors.red
+                    : Colors.green,
+              ),
+              Text(
+                liveMatch.fixture!.status!.short!,
```
Comment (MBanaszynski) [inline lib/app/custom_widgets/match_league_and_timer_widget.dart:67]: Are you sure it will never be null?

File: lib/app/features/home/home_tabs/score_tab/score_tab_views/score_tab_loaded_view.dart [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+    const String stringPlaceholder = '';
+    final String flag = liveMatches[0].league?.flag ?? stringPlaceholder;
+    final String leagueName = liveMatches[0].league?.name ?? stringPlaceholder;
+    
+    if (liveMatches.isEmpty) {
+      return const SizedBox(
+        height: 200,
+        child: Center(
+          child: Text(
+            'No matches were played on this date.',
```
Comment (MBanaszynski) [inline lib/app/features/home/home_tabs/score_tab/score_tab_views/score_tab_loaded_view.dart]: Add to localize

File: lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart [hunk 18]
```diff
@@ -6,9 +6,7 @@ import 'package:score_live/core/enums.dart';
 import 'package:score_live/presentation/constants/app_colors.dart';
 
 class HomeOptionsTapBar extends StatelessWidget {
-  HomeOptionsTapBar({super.key});
-
-  final homeCubit = Modular.get<HomeCubit>();
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart:11]: You could replace `Modular.get` with `context.read`
e.g.
`HomeCubit _homeCubit(BuildContext context) => context.read<HomeCubit>();`

And then in methods `_homeCubit(context).`

File: lib/app/features/home/home_widgets/live_now/cubit/live_now_cubit.dart [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+class LiveNowCubit extends Cubit<LiveNowState> {
+  LiveNowCubit(this.homeScreenRepository) : super(const LoadingMatchesState());
+
+  final HomeScreenRepository homeScreenRepository;
+
+  Future<void> fetchLiveMatches() async {
+    emit(const LoadingMatchesState());
+    try {
+      final liveMatches = await homeScreenRepository.fetchLiveMatches();
+      emit(MatchesLoadedState(liveMatches!));
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/live_now/cubit/live_now_cubit.dart]: Are you sure that live matches will never be null?

File: lib/app/features/home/home_widgets/live_now/live_now_views/live_now_loaded_view.dart [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+  final List<LiveMatchModel> liveMatches;
+
+  @override
+  Widget build(BuildContext context) {
+    if (liveMatches.isEmpty) {
+      return const SizedBox(
+        height: 200,
+        child: Center(
+          child: Text(
+            'Nothing here :(',
```
Comment (MBanaszynski) [inline lib/app/features/home/home_widgets/live_now/live_now_views/live_now_loaded_view.dart]: Add to localize

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart [hunk 21]
```diff
@@ -0,0 +1,166 @@
...
+                      for (final event in matchEvents) ...[
+                        if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
+                          if (homeTeamID == event.team?.id && event.time!.elapsed! < 45)
+                            Padding(
+                              padding: const EdgeInsets.only(left: 8.0, top: 8.0),
+                              child: HomeEventTextSample(
+                                  time: event.time?.elapsed.toString() ?? '', player: event.player?.name ?? ''),
+                            )
+                          else
+                            const SizedBox(height: 0),
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: Consider `SizedBox.shrink()`

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart [hunk 22]
```diff
@@ -0,0 +1,166 @@
...
+                      for (final event in matchEvents) ...[
+                        if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
+                          if (awayTeamID == event.team?.id && event.time!.elapsed! < 45)
+                            Padding(
+                              padding: const EdgeInsets.only(right: 8.0, top: 8.0),
+                              child: AwayEventTextSample(
+                                  time: event.time?.elapsed.toString() ?? '', player: event.player?.name ?? ''),
+                            )
+                          else
+                            const SizedBox(height: 0),
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: Same here

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart [hunk 23]
```diff
@@ -0,0 +1,166 @@
...
+                      for (final event in matchEvents) ...[
+                        if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
+                          if (homeTeamID == event.team?.id && event.time!.elapsed! > 45)
+                            Padding(
+                              padding: const EdgeInsets.only(left: 8.0, top: 8.0),
+                              child: HomeEventTextSample(
+                                  time: event.time?.elapsed.toString() ?? '', player: event.player?.name ?? ''),
+                            )
+                          else
+                            const SizedBox(height: 0),
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: Same here

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart [hunk 24]
```diff
@@ -0,0 +1,166 @@
...
+                      for (final event in matchEvents) ...[
+                        if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
+                          if (awayTeamID == event.team?.id && event.time!.elapsed! > 45)
+                            Padding(
+                              padding: const EdgeInsets.only(right: 8.0, top: 8.0),
+                              child: AwayEventTextSample(
+                                  time: event.time?.elapsed.toString() ?? '', player: event.player?.name ?? ''),
+                            )
+                          else
+                            const SizedBox(height: 0),
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: Same here

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart [hunk 25]
```diff
@@ -0,0 +1,90 @@
...
+
+class FirstHalfResultDisplay extends StatelessWidget {
+  const FirstHalfResultDisplay({super.key, required this.liveMatch});
+
+  final LiveMatchModel liveMatch;
+
+  @override
+  Widget build(BuildContext context) {
+    final String homeTeamName = liveMatch.teams?.home?.name ?? 'Unknown home team';
+    final String awayTeamName = liveMatch.teams?.away?.name ?? 'Unknown away team';
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart]: Add to localize

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+class SecondfHalfResultDisplay extends StatelessWidget {
+  const SecondfHalfResultDisplay({super.key, required this.liveMatch});
+
+  final LiveMatchModel liveMatch;
+
+  @override
+  Widget build(BuildContext context) {
+    final String homeTeamName = liveMatch.teams?.home?.name ?? 'Unknown home team';
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart]: Use localize

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart [hunk 27]
```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:score_live/core/applocalization_context.dart';
+import 'package:score_live/models/live_match_model.dart';
+
+class FirstHalfResultDisplay extends StatelessWidget {
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart]: Do you need two separated widgets instead of passing parameter `isHalfTime` ?

File: lib/app/features/home/match_details/match_details_widgets/match_details_tab_bar.dart [hunk 28]
```diff
@@ -10,8 +9,6 @@ class MatchDetailsTabBar extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final homeCubit = Modular.get<HomeCubit>();
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_widgets/match_details_tab_bar.dart:13]: Same here

File: lib/app/features/home/match_details/match_details_widgets/match_goals_widget.dart [hunk 29]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        final matchEvents = state.matchEvents;
+
+        if (matchEvents.isEmpty) {
+          return const SizedBox(
+            height: 150,
+            child: Center(
+              child: Text(
+                'No details about this match',
```
Comment (MBanaszynski) [inline lib/app/features/home/match_details/match_details_widgets/match_goals_widget.dart]: Add to localize

**Review by MBanaszynski — CHANGES_REQUESTED** [review]

File: lib/app/custom_widgets/match_league_and_timer_widget.dart (hunk 16 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/match_league_and_timer_widget.dart:67]: fixed

File: lib/app/features/home/home_tabs/score_tab/score_tab_views/score_tab_loaded_view.dart (hunk 17 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_tabs/score_tab/score_tab_views/score_tab_loaded_view.dart]: done

File: lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart (hunk 18 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart:11]: done

File: lib/app/features/home/home_widgets/live_now/cubit/live_now_cubit.dart (hunk 19 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_widgets/live_now/cubit/live_now_cubit.dart]: fixed

File: lib/app/features/home/home_widgets/live_now/live_now_views/live_now_loaded_view.dart (hunk 20 above)
Comment (kubaolszewski) [reply lib/app/features/home/home_widgets/live_now/live_now_views/live_now_loaded_view.dart]: done

File: lib/app/features/home/match_details/match_details_widgets/match_details_tab_bar.dart (hunk 28 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_widgets/match_details_tab_bar.dart:13]: done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart (hunk 26 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart]: done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart (hunk 25 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart]: done

File: lib/app/features/home/match_details/match_details_widgets/match_goals_widget.dart (hunk 29 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_widgets/match_goals_widget.dart]: done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart (hunk 27 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_results_display.dart]: done with boolean parameter as suggested

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart (hunk 21 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart (hunk 22 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart (hunk 23 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: done

File: lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart (hunk 24 above)
Comment (kubaolszewski) [reply lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/halves_events_display.dart.dart]: done

**Review by MBanaszynski — APPROVED** [review]
